// dotnav/lib/Dotnav.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Dotnav = function (_React$Component) {
  _inherits(Dotnav, _React$Component);

  function Dotnav() {
    _classCallCheck(this, Dotnav);

    return _possibleConstructorReturn(this, (Dotnav.__proto__ || Object.getPrototypeOf(Dotnav)).apply(this, arguments));
  }

  _createClass(Dotnav, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          withArrows = _props.withArrows,
          count = _props.count,
          index = _props.index,
          classes = _props.classes,
          handleChange = _props.handleChange;


      if (count < 2) return null;

      return _react2.default.createElement(
        'nav',
        { className: classes.root },
        withArrows && _react2.default.createElement('button', {
          type: 'button',
          className: classes.arrowPrev,
          onClick: function onClick() {
            return handleChange(index <= 0 ? count - 1 : index - 1);
          }
        }),
        _react2.default.createElement(
          'ul',
          { className: classes.dots },
          Array.from({ length: count }, function (_, i) {
            return _react2.default.createElement('li', {
              key: i,
              className: index === i ? classes.dotActive : classes.dot,
              onClick: function onClick() {
                return handleChange(i);
              }
            });
          })
        ),
        withArrows && _react2.default.createElement('button', {
          type: 'button',
          className: classes.arrowNext,
          onClick: function onClick() {
            return handleChange(index >= count - 1 ? 0 : index + 1);
          }
        })
      );
    }
  }]);

  return Dotnav;
}(_react2.default.Component);

exports.default = Dotnav;


Dotnav.propTypes = {
  withArrows: _propTypes2.default.bool,
  count: _propTypes2.default.number,
  index: _propTypes2.default.number,
  handleChange: _propTypes2.default.func,
  classes: _propTypes2.default.shape({
    root: _propTypes2.default.string,
    dots: _propTypes2.default.string,
    dot: _propTypes2.default.string,
    dotActive: _propTypes2.default.string,
    arrowPrev: _propTypes2.default.string,
    arrowNext: _propTypes2.default.string
  })
};

Dotnav.defaultProps = {
  withArrows: false,
  count: 0,
  index: 0,
  handleChange: function handleChange() {},
  classes: {
    root: 'dotnav',
    dots: 'dotnav-dots',
    dot: 'dotnav-dot',
    dotActive: 'dotnav-dot dotnav-dot_active',
    arrowPrev: 'dotnav-arrow dotnav-arrow_prev',
    arrowNext: 'dotnav-arrow dotnav-arrow_next'
  }
};
module.exports = exports['default'];
